lib/export/formatters/dataElementFormatterCimpl6.js

```diff
@@ -0,0 +1,688 @@
+
+    if (formatAsField) cardString = rightPad(cardString, this.maxRightPad);
+    cardString = cardString.replace(':', '');
+
+    return `${formattedValue} ${cardString}`;
+  }
+
+  formatValueWithConstraints(formattedByPath, value, de, formatAsField) {
+    let counter = 0;
+    let formattedWithConstraintTerms = [];
+    let formattedWithConstraint = '';
+    let substitutionHash = {};
+    for (const con of value.constraints) {
+      if (con.lastModifiedBy.fqn == de.identifier.fqn) {
+        counter += 1;
+        if (con.constructor.name === 'IncludesTypeConstraint') {
+          if (counter > 1) {
+            formattedWithConstraint = this.formatValueByConstraintType('', con, formatAsField, value);
+          }
+          else {
+            formattedWithConstraint = this.formatValueByConstraintType(formattedByPath, con, formatAsField, value);
+          }
+        }
+        else {
+          formattedWithConstraint = this.formatValueByConstraintType(formattedByPath, con, formatAsField, value);
+        }
+        formattedWithConstraintTerms.push(this.replaceValue(formattedWithConstraint));
+      }
+    }
+    if (formattedWithConstraintTerms.length > 0) {
+      return formattedWithConstraintTerms.join('\n');
+    }
+    else {
+      return formattedByPath;
+    }
+  }
+
+  formatValueByConstraintType(formattedValue, constraint, formatAsField, value) {
+    formattedValue = this.replaceValue(formattedValue);
+    switch (constraint.constructor.name) {
+    case 'TypeConstraint': {
+      if (formatAsField && !value.inheritance && !constraint.onValue) {
+        constraint.onValue = true;
+      }
+      return this.formatValueWithTypeConstraint(formattedValue, constraint);
+    }
+    case 'ValueSetConstraint': {
+      if (!formatAsField && constraint.path.length >= 2 && constraint.path[0].name == 'Units' && constraint.path[1].name == 'Coding') {
+        return [
+          formattedValue,
+          `${rightPad('', this.maxRightPad)}${this.formatValueWithValueSetConstraint(`${formattedValue.replace(/.*\s([^\s]*)/,'$1')}.Units`, constraint)}`,
+        ].join('\r\n');
+      }
+
+      return this.formatValueWithValueSetConstraint(formattedValue, constraint);
+    }
+    case 'CardConstraint': {
+      if (formatAsField) {
+        return this.formatValueWithCardConstraint(formattedValue, constraint);
+      } else {
+        return formattedValue;
+      }
+    }
+    case 'IncludesTypeConstraint':
+      return this.formatValueWithIncludesTypeConstraint(formattedValue, constraint);
+    case 'IncludesCodeConstraint':
+      return this.formatValueWithIncludesCodeConstraint(formattedValue, constraint);
+    case 'BooleanConstraint':
+      return this.formatValueWithBooleanConstraint(formattedValue, constraint);
+    case 'CodeConstraint':
+      return this.formatValueWithCodeConstraint(formattedValue, constraint);
+    default:
+      console.log('learn to deal with this %s', constraint.constructor.name);
```

Comment:
We really shouldn't use `console`.  If we need to log messages, we should probably introduce the Bunyan logger to this code.